src/mavsdk/plugins/telemetry/telemetry_impl.cpp

```diff
@@ -355,7 +355,7 @@ Telemetry::Result TelemetryImpl::set_rate_gps_info(double rate_hz)
 Telemetry::Result TelemetryImpl::set_rate_battery(double rate_hz)
 {
     return telemetry_result_from_command_result(
-        _parent->set_msg_rate(MAVLINK_MSG_ID_SYS_STATUS, rate_hz));
+        _parent->set_msg_rate(MAVLINK_MSG_ID_BATTERY_STATUS, rate_hz));
```

Comment:
This message change was recently merged into main via PR https://github.com/mavlink/MAVSDK/pull/2021 . Maybe rebase current PR and attempt to merge again?